waspc/tools/make_binary_package.sh

```diff
@@ -9,15 +11,19 @@
 DST=$PWD/${1:-wasp.tar.gz}
 
 TMP_DIR="$(mktemp -d 2> /dev/null || mktemp -d -t wasp-bin-package)"
+CABAL_PROJECT_ROOT_PATH="$(cabal list-bin wasp-cli | sed s/\\/dist-newstyle.*//)"
 
 WASP_BINARY_PATH="$(cabal list-bin wasp-cli)"
 cp "$WASP_BINARY_PATH" "$TMP_DIR/wasp-bin"
 
-CABAL_PROJECT_ROOT_PATH="$(cabal list-bin wasp-cli | sed s/\\/dist-newstyle.*//)"
-cp -R "$CABAL_PROJECT_ROOT_PATH/data" "$TMP_DIR/data"
+cabal sdist --list-only \
+  | grep "^./data" \
+  | while read -r data_file_path; do
+    mkdir -p "$TMP_DIR/$(dirname "$data_file_path")"
+    cp "$CABAL_PROJECT_ROOT_PATH/$data_file_path" "$TMP_DIR/$data_file_path"
+  done
```

Comment:
Can we make it more obvious what this piece of code does with some nicely named functions and/or variables (or, if all else fails, some comments).

Btw, I think this would possibly be clearer and less fragile with `rsync`, not sure.